app/backend/app.py

```diff
@@ -83,10 +83,57 @@ async def assets(path):
     return await send_from_directory(Path(__file__).resolve().parent / "static" / "assets", path)
 
 
+# Decorator for routes that request a specific file that might require access control enforcement
+def authenticated_path(route_fn: Callable[[str], Any]):
+    async def auth_handler(path=""):
+        # If authentication is enabled, validate the user can access the file
+        auth_helper = current_app.config[CONFIG_AUTH_CLIENT]
+        search_client = current_app.config[CONFIG_SEARCH_CLIENT]
+        authorized = False
+        try:
+            auth_claims = await auth_helper.get_auth_claims_if_enabled(request.headers)
+            authorized = await auth_helper.check_path_auth(path, auth_claims, search_client)
+        except AuthError:
+            abort(403)
+        except Exception as error:
+            logging.exception("Problem checking path auth %s", error)
+            return error_response(error, route="/content")
+
+        if not authorized:
+            abort(403)
+
+        return await route_fn(path)
+
+    # Rename decorator to avoid view function mapping overwrite error
+    auth_handler.__name__ = route_fn.__name__
+
+    return auth_handler
+
+
+# Decorator for routes that might require access control. Unpacks Authorization header informaiton into an auth_claims dictionary
+def authenticated(route_fn: Callable[[Dict[str, Any]], Any]):
+    async def auth_handler():
+        auth_helper = current_app.config[CONFIG_AUTH_CLIENT]
+        try:
+            auth_claims = await auth_helper.get_auth_claims_if_enabled(request.headers)
+        except AuthError:
+            abort(403)
+
+        return await route_fn(auth_claims)
+
+    # Rename decorator to avoid view function mapping overwrite error
+    auth_handler.__name__ = route_fn.__name__
+
+    return auth_handler
+
+
 # Serve content files from blob storage from within the app to keep the example self-contained.
-# *** NOTE *** this assumes that the content files are public, or at least that all users of the app
-# can access all the files. This is also slow and memory hungry.
+# *** NOTE *** if you are using app services authentication, this route will return unauthorized to all users that are not logged in
+# if AZURE_ENFORCE_ACCESS_CONTROL is not set or false, logged in users can access all files regardless of access control
+# if AZURE_ENFORCE_ACCESS_CONTROL is set to true, logged in users can only access files they have access to
+# This is also slow and memory hungry.
```

Comment:
Ditto for these other function comments, here and above. Another advantage of docstrings is that if someone collapses a function to declutter up their working environment, the docstring collapses with it.